lts_array/lts_array_plot.py

```diff
@@ -29,86 +29,105 @@ def lts_array_plot(st, lts_vel, lts_baz, t, mdccm, stdict):
             ``axs1``: Output figure axes.
 
     Example:
-        fig1, axs1 = lts_array_plot(st, stdict, t, mdccm, lts_vel, lts_baz)
+        fig, axs = lts_array_plot(st, lts_vel, lts_baz, t, mdccm, stdict)
     '''
 
     # Specify the colormap.
-    CM = 'RdYlBu_r'
+    cm = 'RdYlBu_r'
     # Colorbar/y-axis limits for MdCCM.
-    CAX = (0.2, 1)
+    cax = (0.2, 1)
     # Specify the time vector for plotting the trace.
     tvec = st[0].times('matplotlib')
 
+    # Check station dictionary input. It must be a non-empy dictionary.
+    if not isinstance(stdict, dict) or not stdict:
+        stdict = None
+
+    # Determine the number and order of subplots.
+    num_subplots = 3
+    vplot = 1
+    bplot = 2
+    splot = bplot
+    if stdict is not None:
+        num_subplots += 1
+        splot = bplot + 1
+
     # Start plotting.
-    fig1, axarr1 = plt.subplots(4, 1, sharex='col')
-    fig1.set_size_inches(9, 12)
-    axs1 = axarr1.ravel()
-    axs1[0].plot(tvec, st[0].data, 'k')
-    axs1[0].axis('tight')
-    axs1[0].set_ylabel('Pressure [Pa]')
-    axs1[0].text(0.15, 0.93, st[0].stats.station, horizontalalignment='center',
-                 verticalalignment='center', transform=axs1[0].transAxes)
-    cbaxes = fig1.add_axes(
-        [0.9, axs1[2].get_position().y0, 0.02,
-         axs1[1].get_position().y1 - axs1[2].get_position().y0])
+    fig, axarr = plt.subplots(num_subplots, 1, sharex='col')
+    fig.set_size_inches(9, 12)
+    axs = axarr.ravel()
+    axs[0].plot(tvec, st[0].data, 'k')
+    axs[0].axis('tight')
+    axs[0].set_ylabel('Pressure [Pa]')
+    axs[0].text(0.15, 0.93, st[0].stats.station, horizontalalignment='center',
+                verticalalignment='center', transform=axs[0].transAxes)
+    cbaxes = fig.add_axes(
+        [0.95, axs[splot].get_position().y0, 0.02,
+         axs[vplot].get_position().y1 - axs[splot].get_position().y0])
 
     # Plot the trace velocity plot.
-    sc = axs1[1].scatter(t, lts_vel, c=mdccm,
-                         edgecolors='gray', lw=0.1, cmap=CM)
-    axs1[1].set_ylim(0.15, 0.60)
-    axs1[1].set_xlim(t[0], t[-1])
-    axs1[1].plot([t[0], t[-1]], [0.25, 0.25], '-', color='grey')
-    axs1[1].plot([t[0], t[-1]], [0.45, 0.45], '-', color='grey')
-    sc.set_clim(CAX)
-    axs1[1].set_ylabel('Trace Velocity\n [km/s]')
-    axs1[1].grid(b=1, which='major', color='gray', linestyle=':', alpha=0.5)
+    sc = axs[vplot].scatter(t, lts_vel, c=mdccm,
+                            edgecolors='gray', lw=0.1, cmap=cm)
+    axs[vplot].set_ylim(0.25, 0.45)
+    axs[vplot].set_xlim(t[0], t[-1])
+    axs[vplot].plot([t[0], t[-1]], [0.25, 0.25], '-', color='grey')
+    axs[vplot].plot([t[0], t[-1]], [0.45, 0.45], '-', color='grey')
+    sc.set_clim(cax)
+    axs[vplot].set_ylabel('Trace Velocity\n [km/s]')
+    axs[vplot].grid(b=1, which='major', color='gray', linestyle=':', alpha=0.5)
```

Comment:
One more: why do you have the gridlines gray? It's different than the top subplot and how we usually plot